docs/events-interception.md

```diff
@@ -0,0 +1,190 @@
+        // this handler will never be called
+    });
+
+    testplane.on(testplane.events.TEST_PENDING, (test) => {
+        // this handler will always be called instead of the handler for TEST_FAIL
+    });
+};
+```
+
+## Leaving the event as is {#leaving-event-as-is}
+
+If for some reason the intercepted event needs to be left _as is_, then its handler must return exactly the same object or any _falsey_ value: _undefined, null_ or _false._
+
+```javascript
+module.exports = (testplane) => {
+    testplane.intercept(testplane.events.TEST_FAIL, ({ event, data }) => {
+        return { event, data };
+        // or return;
+        // or return null;
+        // or return false;
+    });
+
+    testplane.on(testplane.events.TEST_FAIL, (test) => {
+        // this handler will be called as usual,
+        // because intercepting the TEST_FAIL event does not change it in any way
+    });
+};
+```
+
+## Ignoring an event {#ignoring-event}
+
+To ignore an event and prevent it from propagating further, you need to return an empty object from the handler (in which the event is intercepted):
+
+```javascript
+module.exports = (testplane) => {
+    testplane.intercept(testplane.events.TEST_FAIL, ({ event, data }) => {
+        return {};
+    });
+
+    testplane.on(testplane.events.TEST_FAIL, (test) => {
+        // this handler will never be called
+    });
+};
+```
+
+## Delaying event processing {#delaying-events}
+
+The above approach with ignoring an event can be used to delay the occurrence of certain events, for example:
+
+```javascript
+module.exports = (testplane) => {
+    const intercepted = [];
+
+    testplane.intercept(testplane.events.TEST_FAIL, ({ event, data }) => {
+        // collect all TEST_FAIL events
+        intercepted.push({ event, data });
+
+        // and do not let them propagate further
+        return {};
+    });
+
+    testplane.on(testplane.events.END, () => {
+        // after the test run is finished, trigger all accumulated TEST_FAIL events
+        intercepted.forEach(({ event, data }) => testplane.emit(event, data));
+    });
+};
+```
+
+### Passing information between event handlers {#passing-information-between-event-handlers}
+
+Events that are triggered in the master process and in Testplane workers cannot exchange information via global variables.
+
+For example, this approach will not work:
+
+```javascript
+module.exports = (testplane) => {
+    let flag = false;
+
+    testplane.on(testplane.events.RUNNER_START, () => {
+        flag = true;
+    });
+
+    testplane.on(testplane.events.NEW_BROWSER, () => {
+        // false will be displayed because the NEW_BROWSER event
+        // is triggered in the testplane worker, and RUNNER_START in the master process
+        console.info(flag);
+    });
+
+    testplane.on(testplane.events.RUNNER_END, () => {
+        // true will be output
+        console.info(flag);
+    });
+};
+```
+
+But you can solve the issue like this:
```

Comment:
Hmm, I'm not 100% sure about this, but isn't `testplane.ctx` the preferred way of passing data around?

Also, I think it would be nice to mention that upon transferring between master and workers objects go trhough serialization/deserialization (in particular, objects methods will be lost)

Reply:
Also you have an example in the new worker event showcasing `send()` API. Maybe we should mention it here as well.

Reply:
Here you have link to example with `NEW_WORKER_PROCESS` `send()`:  https://github.com/gemini-testing/testplane/pull/943/files/2ced168569b3262137a094bed331b763805a1268#diff-c3771283669161fa09974fac2f11e96726f30a591d2f3f52e6770ed6e9d3fb19R142